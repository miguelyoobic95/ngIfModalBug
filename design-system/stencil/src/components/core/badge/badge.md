---
name: Badge
category: Core Components
---

Tags can be used with `<yoo-badge>` elements.

## Interface

```tsx
    export interface IBadgeEntry {
        text?: string;
        closable?: boolean;
        iconLeft?: string;
        iconRight?: string;
}
```
s
## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|""|the text inside the badge|
|`closable`|boolean|false|make the badge is closable |
|`iconLeft`|string||icon to add to on the left-hand-side of the badge|
|`iconRight`|string||icon to add to on the right-hand-side of the badge|

```yoo-badge-new.html
    <yoo-badge text="Sat 29 Jun" closable="true" icon-left="yo-home" class="large outline"></yoo-badge>
    <br>
    <br>    
    <yoo-badge text="Sat 29 Jun" icon-right="yo-home" icon-left="yo-close" class="outline"></yoo-badge>
    <br>
    <br>
    <yoo-badge text="Sat 29 Jun" icon-right="yo-home" icon-left="yo-close" class="small outline"></yoo-badge>
    <br>
    <yoo-badge text="Sat 29 Jun" icon-right="yo-home" icon-left="yo-close" class="small outline success"></yoo-badge>
     <br>
    <yoo-badge text="Sat 29 Jun" closable="true" icon-left="yo-home" class="large outline success"></yoo-badge>
```

```yoo-label.html
<div style="padding:10px;">
    <yoo-badge text="Mission" class="round medium gradient-accent"></yoo-badge>
    <yoo-badge text="Approved" class="round medium gradient-dark"></yoo-badge>
    <yoo-badge text="Mission" class="round medium gradient-success"></yoo-badge>
    <yoo-badge text="Rejected" class="round medium gradient-danger"></yoo-badge>
    <yoo-badge text="Ongoing" class="round medium gradient-info"></yoo-badge>
    <yoo-badge text="Pending" class="round medium gradient-warning"></yoo-badge>
</div>
```
### Text Attribute

```yoo-label-text.html
<div style="padding:10px;">
    <yoo-badge text="Tag" class="accent"></yoo-badge>
</div>
```

### Icon Attribute

```yoo-label-text.html
<div style="padding:10px;">
    <yoo-badge icon="yo-fire" text="Fire" class="danger"></yoo-badge>
</div>
```

### Closeable Attribute

```yoo-label-closeable.html
<div style="padding:10px;">
    <yoo-badge text="Tag success closable outline small" closable="true" class="success outline small"></yoo-badge>
    <yoo-badge text="Tag success closable outline" closable="true" class="success outline"></yoo-badge>
    <yoo-badge text="Tag success closable" closable="true" class="success"></yoo-badge>
    <br>
    <yoo-badge text="London HQ" closable="true" class="outline large"></yoo-badge>
    
</div>
```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|label accent|
|Dark|`dark`|label dark|
|Success|`success`|label success|
|Danger|`danger`|label danger|
|Info|`info`|label info|
|Warning|`warning`|label warning|
|Gradient Accent|`gradient-accent`|label gradient accent|
|Gradient Danger|`gradient-danger`|label gradient danger|
|Gradient Info|`gradient-info`|label gradient info|
|Gradient Warning|`gradient-warning`|label gradient warning|
|Gradient Success|`gradient-success`|label gradient success|
|Gradient Dark|`gradient-dark`|label gradient dark|
|Link|`link`|label for link with underline on hover|
|Outline|`outline`|label outlined|
|Medium|`medium`|medium label (18px)|
|Small|`small`|small label (14px)|
|Round|`round`|round label|
|Transparent|`transparent`|transparent label|



### Outline Class

```yoo-badge-outline.html
<div style="padding:10px;">
    <yoo-badge text="Tag accent outline" class="accent outline"></yoo-badge>
    <yoo-badge text="Tag dark outline" class="dark outline"></yoo-badge>
    <yoo-badge text="Tag success outline" class="success outline"></yoo-badge>
    <yoo-badge text="Tag danger outline" class="danger outline"></yoo-badge>
    <yoo-badge text="Tag info outline" class="info outline"></yoo-badge>
    <yoo-badge text="Tag warning outline" class="warning outline"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag accent" class="accent"></yoo-badge>
    <yoo-badge text="Tag dark" class="dark"></yoo-badge>
    <yoo-badge text="Tag success" class="success"></yoo-badge>
    <yoo-badge text="Tag danger" class="danger"></yoo-badge>
    <yoo-badge text="Tag info" class="info"></yoo-badge>
    <yoo-badge text="Tag warning" class="warning"></yoo-badge>
</div>
```


### Size Variations

```yoo-badge.html
<div style="padding:10px;">
    <yoo-badge text="Tag small accent outline" class="small accent outline"></yoo-badge>
    <yoo-badge text="Tag accent outline" class="accent outline"></yoo-badge>
</div>
```

```yoo-badge.css hidden
    yoo-badge {
        margin-bottom: 5px;
        display: inline-block;
    }
```

### Style Variations

```yoo-badge-style.html
<div style="padding:10px;">
    <yoo-badge text="Tag accent outline" class="accent outline"></yoo-badge>
    <yoo-badge text="Tag dark outline" class="dark outline"></yoo-badge>
    <yoo-badge text="Tag success outline" class="success outline"></yoo-badge>
    <yoo-badge text="Tag danger outline" class="danger outline"></yoo-badge>
    <yoo-badge text="Tag info outline" class="info outline"></yoo-badge>
    <yoo-badge text="Tag warning outline" class="warning outline"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag gradient-accent outline" class="gradient-accent outline"></yoo-badge>
    <yoo-badge text="Tag gradient-dark outline" class="gradient-dark outline"></yoo-badge>
    <yoo-badge text="Tag gradient-success outline" class="gradient-success outline"></yoo-badge>
    <yoo-badge text="Tag gradient-danger outline" class="gradient-danger outline"></yoo-badge>
    <yoo-badge text="Tag gradient-info outline" class="gradient-info outline"></yoo-badge>
    <yoo-badge text="Tag gradient-warning outline" class="gradient-warning outline"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag gradient-accent" class="gradient-accent"></yoo-badge>
    <yoo-badge text="Tag gradient-dark" class="gradient-dark"></yoo-badge>
    <yoo-badge text="Tag gradient-success" class="gradient-success"></yoo-badge>
    <yoo-badge text="Tag gradient-danger" class="gradient-danger"></yoo-badge>
    <yoo-badge text="Tag gradient-info" class="gradient-info"></yoo-badge>
    <yoo-badge text="Tag gradient-warning" class="gradient-warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag danger link outline small" class="danger link outline small"></yoo-badge>
    <yoo-badge text="Tag danger link outline" class="danger link outline"></yoo-badge>
    <yoo-badge text="Tag danger link" class="danger link"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag gradient-accent" class="round gradient-accent"></yoo-badge>
    <yoo-badge text="Tag gradient-dark" class="round gradient-dark"></yoo-badge>
    <yoo-badge text="Tag gradient-success" class="round gradient-success"></yoo-badge>
    <yoo-badge text="Tag gradient-danger" class="round gradient-danger"></yoo-badge>
    <yoo-badge text="Tag gradient-info" class="round gradient-info"></yoo-badge>
    <yoo-badge text="Tag gradient-warning" class="round gradient-warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag gradient-accent" class="round medium gradient-accent"></yoo-badge>
    <yoo-badge text="Tag gradient-dark" class="round medium gradient-dark"></yoo-badge>
    <yoo-badge text="Tag gradient-success" class="round medium gradient-success"></yoo-badge>
    <yoo-badge text="Tag gradient-danger" class="round medium gradient-danger"></yoo-badge>
    <yoo-badge text="Tag gradient-info" class="round medium gradient-info"></yoo-badge>
    <yoo-badge text="Tag gradient-warning" class="round medium gradient-warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag gradient-accent" class="round small gradient-accent"></yoo-badge>
    <yoo-badge text="Tag gradient-dark" class="round small gradient-dark"></yoo-badge>
    <yoo-badge text="Tag gradient-success" class="round small gradient-success"></yoo-badge>
    <yoo-badge text="Tag gradient-danger" class="round small gradient-danger"></yoo-badge>
    <yoo-badge text="Tag gradient-info" class="round small gradient-info"></yoo-badge>
    <yoo-badge text="Tag gradient-warning" class="round small gradient-warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="Tag outline accent" class="round outline accent"></yoo-badge>
    <yoo-badge text="Tag outline dark" class="round outline dark"></yoo-badge>
    <yoo-badge text="Tag outline success" class="round outline success"></yoo-badge>
    <yoo-badge text="Tag outline danger" class="round outline danger"></yoo-badge>
    <yoo-badge text="Tag outline info" class="round outline info"></yoo-badge>
    <yoo-badge text="Tag outline warning" class="round outline warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge icon="yo-fire" text="Tag outline accent" class="round outline accent"></yoo-badge>
    <yoo-badge icon="yo-fire" text="Tag outline dark" class="round outline dark"></yoo-badge>
    <yoo-badge icon="yo-fire" text="Tag outline success" class="round outline success"></yoo-badge>
    <yoo-badge icon="yo-fire" text="Tag outline danger" class="round outline danger"></yoo-badge>
    <yoo-badge icon="yo-fire" text="Tag outline info" class="round outline info"></yoo-badge>
    <yoo-badge icon="yo-fire" text="Tag outline warning" class="round outline warning"></yoo-badge>
    <br/>
    <br/>
    <yoo-badge text="TRANSPARENT" class="transparent"></yoo-badge>
    <yoo-badge icon="yo-fire" text="TRANSPARENT" class="transparent"></yoo-badge>
    <yoo-badge icon="yo-fire" text="ROUND TRANSPARENT" class="round transparent"></yoo-badge>
</div>
```

## Events

|Attr|Description|
|---|---|---|
|`tagClosed`|event emitted when the tag is closed|