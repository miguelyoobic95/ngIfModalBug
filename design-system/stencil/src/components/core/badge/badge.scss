

@mixin color-class($class, $color){
    :host(.#{$class}) {
        .outer-container {
            background: cssvar($color);
            .inner-container {
                color: cssvar(light);
                .inner-text {
                    color: cssvar(light);  
                }
            }
        }
    }
}

@mixin outline-color-class($class, $color, $gradient: none){
    @if($gradient == none){
        :host(.#{$class}.outline){
            .outer-container {
                border-color: cssvar($color);
                .inner-container {
                    color: cssvar($color);
                    .inner-text {
                        color: cssvar($color);
                    }
                }
            }
        }
    } @else {
        :host(.#{$class}.outline){
            .outer-container {
                border-image: cssvar($gradient);
                border-image-slice: 1;
                .inner-container {
                    color: cssvar($color);
                }
                .inner-text {
                    color: cssvar($color) !important;
                }
            }
        }
    }
}

:host {
    display: inline-block;
    font-weight: 400;    
    .outer-container {
        transition: all 0.3s;
        opacity: 1;
        border-radius: 2px;
        display: inline-block;
        .inner-container {
            align-items: center;
            justify-content: center;
            line-height: normal;
            padding: 0.25rem 0.625rem;
            font-size: 0.75rem;
            .inner-text {
                color: cssvar(stable);
            }
            .icon-close {
                padding-left: 0.375rem;
                cursor: pointer;
            }
            .icon-left {
                padding-right: 0.375rem;
            }
            .icon-right {
                padding-left: 0.375rem;
            }
        }
        &.closed {
            opacity: 0;
            //display: none;
        }
    }
}

:host(.round) {
    .outer-container {
        border-radius: 2rem;
    }
}

:host(.uppercase) {
    text-transform: uppercase;    
}
:host(.small) {
    .outer-container {
        .inner-container{
            font-size: 0.5625rem;
        }
    }
}

:host(.large) {
    .outer-container {
        // min-width: 140px;
        min-height: 30px;
        .inner-container {
            font-size: 0.9375rem;
            // padding: 5px 8px;
            .icon-close {
                padding-left: 0.75rem;
            }
            .icon-left {
                padding-right: 0.75rem;
            }
            .icon-right {
                padding-left: 0.75rem;
            }
    }
  }
}

:host(.link){
    cursor: pointer;
    .inner-container:hover{
        text-decoration: underline; 
    }
}

:host(.outline){
    &.large {
        .outer-container {
            .inner-container {
                // Override standard padding
                padding: 0.25rem 0.5rem;
            }
        }
    }
    .outer-container {
        background: cssvar(light) !important;
        border: solid 1px rgba(40, 47, 54, 0.15);
        border-radius: 5px;
        .inner-container {
            // Override standard padding
            padding: 0.25rem 0.25rem;
        }
    }
}

:host(.transparent) {
    .outer-container {
        background: rgba(255, 255, 255, 0.4);
        color: cssvar(light);
    }
}

:host(.notification) {
    .outer-container {
        border-radius: 2rem;
        .inner-container {
            font-size: 0.375rem;            
            padding: 0.21875rem 0.34375rem;
        }
    }
}

:host(.notification-medium) {
    .outer-container {
        border-radius: 2rem;
        .inner-container {
            font-size: 0.6875rem;            
            padding: 0.15625rem 0.34375rem;
            transform: translateY(-0.0625rem);
        }
    }
}

@include color-class(accent,accent);
@include color-class(dark,dark);
@include color-class(success,success);
@include color-class(danger,danger);
@include color-class(info,info);
@include color-class(warning,warning);
@include color-class(gradient-warning,gradient-warning);
@include color-class(gradient-success,gradient-success);
@include color-class(gradient-accent,gradient-accent);
@include color-class(gradient-info,gradient-info);
@include color-class(gradient-dark,gradient-dark);
@include color-class(gradient-danger,gradient-danger);

@include outline-color-class(accent, accent);
@include outline-color-class(danger, danger);
@include outline-color-class(warning, warning);
@include outline-color-class(success, success);
@include outline-color-class(info, info);
@include outline-color-class(dark, dark);
@include outline-color-class(gradient-warning, warning, gradient-warning);
@include outline-color-class(gradient-success, success, gradient-success);
@include outline-color-class(gradient-dark, dark, gradient-dark);
@include outline-color-class(gradient-info, info, gradient-info);
@include outline-color-class(gradient-accent, accent, gradient-accent);
@include outline-color-class(gradient-danger, danger, gradient-danger);
