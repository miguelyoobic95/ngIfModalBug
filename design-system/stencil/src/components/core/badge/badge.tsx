import { Component, Prop, Event, EventEmitter, State, Element } from '@stencil/core';

@Component({
    tag: 'yoo-badge',
    styleUrl: 'badge.scss',
    scoped: true
})
export class YooBadgeComponent {

    @Prop() text: string;
    @Prop() closable: boolean;
    @Prop() iconLeft: string;
    @Prop() iconRight: string;

    @Event() tagClosed: EventEmitter<boolean>;
    @Event() rightIconClicked: EventEmitter<boolean>;
    @Event() leftIconClicked: EventEmitter<boolean>;

    @State() closed: boolean = false;

    @Element() host: HTMLStencilElement;

    onClose(): void {
        this.tagClosed.emit(true);
        this.closed = true;
    }

    onLeftIconClicked(): void {
        this.leftIconClicked.emit(true);
    }

    onRightIconClicked(): void {
        this.rightIconClicked.emit(true);
    }

    render(): JSX.Element {
        return (
            <div class={'outer-container' + ((this.closed) ? ' closed' : '')}>
                <div class="inner-container" attr-layout="row">
                    {this.iconLeft ? <i class={'icon-left ' + this.iconLeft} onClick={this.onLeftIconClicked.bind(this)}></i> : null }
                    {this.text ? <span class="inner-text">{this.text}</span> : null}
                    {this.iconRight ? <i class={'icon-right ' + this.iconRight} onClick={this.onRightIconClicked.bind(this)}></i> : null}
                    {this.closable ? <i class="icon-close yo-close" onClick={this.onClose.bind(this)}></i> : null}
                </div>
            </div>
        );
    }
}
