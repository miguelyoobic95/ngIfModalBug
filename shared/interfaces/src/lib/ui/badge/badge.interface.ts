export interface IBadgeEntry {
    text?: string;
    closable?: boolean;
    iconLeft?: string;
    iconRight?: string;
    cssClass?: string;
}