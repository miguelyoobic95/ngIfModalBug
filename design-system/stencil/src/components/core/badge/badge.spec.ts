import { TestWindow } from '@stencil/core/testing';
import { YooBadgeComponent } from './badge';

fdescribe('YooBadgeComponent', () => {
    let window;
    beforeEach(() => {
      window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooBadgeComponent()).toBeTruthy();
    });
    describe('Rendering', () => {
        it ('Should display text', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="34"></yoo-badge>'
              });
            expect(element).toMatchSnapshot();
        });

        it ('Should be displayed with the closeable icon', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="34" closeable="true"></yoo-badge>'
              });
            expect(element).toMatchSnapshot();
        });

        it ('Should be displayed with an icon', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="34" icon="yo-fire"></yoo-badge>'
              });
            expect(element).toMatchSnapshot();
        });
    });

    describe('Re-Rendering', () => {
        it('Should re-render when text is updated', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="Tag"></yoo-badge>'
             });
             expect(element.text).toEqual('Tag');
             element.text = 'Hey';
             await window.flush();
             expect(element.text).toEqual('Hey');
             expect(element).toMatchSnapshot();
        });

        it('Should re-render when closeable is updated', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="Tag"></yoo-badge>'
             });
             element.closeable = true;
             await window.flush();
             expect(element.closeable).toEqual(true);
             expect(element).toMatchSnapshot();
        });

        it('Should re-render when icon is updated', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge icon="yo-fire" text="Tag"></yoo-badge>'
             });
             element.icon = 'yo-flag';
             await window.flush();
             expect(element.icon).toEqual('yo-flag');
             expect(element).toMatchSnapshot();
        });
    });

    describe('Events', () => {
        it('Should trigger an event when closed', async () => {
            let element = await window.load({
                components: [YooBadgeComponent],
                html: '<yoo-badge text="Tag"></yoo-badge>'
             });
             element.closeable = true;
             expect(element.closeable).toEqual(true);
             await window.flush();
             element.close = jest.fn();
             await element.close();
             expect(element.close).toBeCalled();
        });
    });
});
