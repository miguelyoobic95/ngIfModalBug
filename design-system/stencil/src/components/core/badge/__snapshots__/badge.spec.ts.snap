// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooBadgeComponent Re-Rendering Should re-render when closeable is updated 1`] = `
<yoo-badge
  class="hydrated"
  data-yoo-badge-host=""
  text="Tag"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        Tag
      </span>
      
    </div>
  </div>
</yoo-badge>
`;

exports[`YooBadgeComponent Re-Rendering Should re-render when icon is updated 1`] = `
<yoo-badge
  class="hydrated"
  data-yoo-badge-host=""
  icon="yo-fire"
  text="Tag"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        Tag
      </span>
      
    </div>
  </div>
</yoo-badge>
`;

exports[`YooBadgeComponent Re-Rendering Should re-render when text is updated 1`] = `
<yoo-badge
  class="hydrated"
  data-yoo-badge-host=""
  text="Tag"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        Hey
      </span>
      
    </div>
  </div>
</yoo-badge>
`;

exports[`YooBadgeComponent Rendering Should be displayed with an icon 1`] = `
<yoo-badge
  class="hydrated"
  data-yoo-badge-host=""
  icon="yo-fire"
  text="34"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        34
      </span>
      
    </div>
  </div>
</yoo-badge>
`;

exports[`YooBadgeComponent Rendering Should be displayed with the closeable icon 1`] = `
<yoo-badge
  class="hydrated"
  closeable="true"
  data-yoo-badge-host=""
  text="34"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        34
      </span>
      
    </div>
  </div>
</yoo-badge>
`;

exports[`YooBadgeComponent Rendering Should display text 1`] = `
<yoo-badge
  class="hydrated"
  data-yoo-badge-host=""
  text="34"
>
  
  <div
    class="outer-container"
    data-yoo-badge=""
  >
    <div
      attr-layout="row"
      class="inner-container"
      data-yoo-badge=""
    >
      
      <span
        class="inner-text"
        data-yoo-badge=""
      >
        34
      </span>
      
    </div>
  </div>
</yoo-badge>
`;
